### PR TITLE
Filled out majority of LDAP type definitions for version 1.x

### PR DESCRIPTION
**Pull request by nickzelei**

### CODE REVIEW

**Comment by unional** [comment]
👍  Please also update the entry in https://github.com/typings/registry

File: lib/client/client.d.ts [hunk 1]
```diff
@@ -89,7 +95,24 @@ declare namespace Client {
...
+    export interface SearchResult {
+        status: number;
+        controls: Control[];
+    }
+
+    export interface SearchResponseEmitter extends EventEmitter {
+        on(ev: string, callback: (entry: SearchEntry) => void): this;
+        on(ev: string, callback: (referral: any) => void): this;
+        on(ev: string, callback: (err: LDAPError) => void): this;
+        on(ev: string, callback: (result: SearchResult) => void): this;
```
Comment (felixfbecker) [inline lib/client/client.d.ts:110]: why have these signatures in addition to the ones below?

**Review by felixfbecker — COMMENTED** [review]
Looks great! I left out all the types I didn't need back then, awesome to see someone picking up the work

File: lib/client/client.d.ts (hunk 1 above)
Comment (nickzelei) [reply lib/client/client.d.ts:110]: Hmm, I'm not entirely sure why I left those in. Looks like an oversight. I deleted them from my local types and my code still compiled.

**Pull request merged**